src/otx/v2/adapters/torch/lightning/visual_prompt/modules/datasets/dataset.py

```diff
@@ -0,0 +1,392 @@
+
+            self.train_dataset = OTXVisualPromptingDataset(
+                train_otx_dataset,
+                image_size,
+                mean,
+                std,
+                offset_bbox=self.config.offset_bbox,
+            )
+            self.val_dataset = OTXVisualPromptingDataset(val_otx_dataset, image_size, mean, std)
+
+        if stage == "test":
+            test_otx_dataset = self.dataset.get_subset(Subset.TESTING)
+            self.test_dataset = OTXVisualPromptingDataset(test_otx_dataset, image_size, mean, std)
+
+        if stage == "predict":
+            predict_otx_dataset = self.dataset
+            self.predict_dataset = OTXVisualPromptingDataset(predict_otx_dataset, image_size, mean, std)
+
+    def summary(self) -> None:
+        """Print size of the dataset, number of images."""
+        for subset in [Subset.TRAINING, Subset.VALIDATION, Subset.TESTING]:
+            dataset = self.dataset.get_subset(subset)
+            num_items = len(dataset)
+            logger.info(
+                "'%s' subset size: Total '%d' images.",
+                subset,
+                num_items,
+            )
+
+    def train_dataloader(self) -> DataLoader | (list[DataLoader] | dict[str, DataLoader]):
+        """Train Dataloader.
+
+        Returns:
+            Union[DataLoader, list[DataLoader], Dict[str, DataLoader]]: Train dataloader.
+        """
+        return DataLoader(
+            self.train_dataset,
+            shuffle=True,
+            batch_size=self.config.train_batch_size,
+            num_workers=self.config.num_workers,
+            collate_fn=collate_fn,
+        )
+
+    def val_dataloader(self) -> DataLoader | list[DataLoader]:
+        """Validation Dataloader.
+
+        Returns:
+            Union[DataLoader, list[DataLoader]]: Validation Dataloader.
+        """
+        return DataLoader(
+            self.val_dataset,
+            shuffle=False,
+            batch_size=self.config.val_batch_size,
+            num_workers=self.config.num_workers,
+            collate_fn=collate_fn,
+        )
+
+    def test_dataloader(self) -> DataLoader | list[DataLoader]:
+        """Test Dataloader.
+
+        Returns:
+            Union[DataLoader, list[DataLoader]]: Test Dataloader.
+        """
+        return DataLoader(
+            self.test_dataset,
+            shuffle=False,
+            batch_size=self.config.test_batch_size,
+            num_workers=self.config.num_workers,
+            collate_fn=collate_fn,
+        )
+
+    def predict_dataloader(self) -> DataLoader | list[DataLoader]:
+        """Predict Dataloader.
+
+        Returns:
+            Union[DataLoader, list[DataLoader]]: Predict Dataloader.
+        """
+        return DataLoader(
+            self.predict_dataset,
+            shuffle=False,
+            batch_size=1,
+            num_workers=self.config.num_workers,
+            collate_fn=collate_fn,
+        )
+
+
+class VisualPromptInferenceDataset(Dataset):
```

Comment:
In fact, the reason I created this is because the current `OTXVisualPromptingDataModule` is based on a `DatasetEntity`.
Instead, only the predict part is rewritten to not use a `DatasetEntity` (`VisualPromptInferenceDataset`).
So currently, this is an OTXVisualPromptingDataModule that can't be reused without DatasetEntity, and if we want to utilize the DataModule, we'll need to create another new DataModule class.

So my suggestion would be to refactor `OTXVisualPromptingDataset`, `OTXVisualPromptingDataModule `and `VisualPromptInferenceDataset` at once in the future (with removing `DatasetEntity`), what do you think?